polynote-kernel/src/main/scala/polynote/kernel/util/RefMap.scala

```diff
@@ -25,53 +25,54 @@ class RefMap[K, V] private (
       semaphore.withPermit {
         ZIO(underlying.containsKey(key)).flatMap {
           case true => ZIO.succeed(underlying.get(key)).flatMap(_.get)
-          case false => create.tap(v => Ref[Task].of(v).flatMap(ref => ZIO(underlying.put(key, ref))))
+          case false => create.tap(v => Ref.make[V](v).flatMap(ref => ZIO.effectTotal(Option(underlying.put(key, ref))).unit))
         }
       }
     }
   }
 
-  def getOrCreateRef[E](key: K)(create: RIO[E, Ref[Task, V]]): RIO[E, V] = synchronized {
```

Comment:
I think `getOrCreate` is a nice thing to have in an API like this - even though I guess we aren't using it now.

Reply:
There's still `getOrCreate`, this was just a method that wasn't used anywhere for if you want to pass your own `Ref` instead of a value. I was changing it from cats `Ref` to ZIO `Ref` so it was either update this unused method or remove it. I say remove (it can always be re-added if a use case comes up)

Reply:
Ahhhh sorry, it was hidden by the fold - my bad! 😞